server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -333,4 +246,125 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
+
+            // Re-evaluate the conditions, now with our final source index name
+            final Map<String, Boolean> postConditionResults = evaluateConditions(
+                rolloverRequest.getConditions().values(),
+                buildStats(currentState.metadata().index(sourceIndexName), statsResponse)
+            );
+            final List<Condition<?>> metConditions = rolloverRequest.getConditions()
+                .values()
+                .stream()
+                .filter(condition -> postConditionResults.get(condition.toString()))
+                .collect(Collectors.toList());
+            // Update the final condition results so they can be used when returning the response
+            conditionResults.set(postConditionResults);
+
+            if (postConditionResults.size() == 0 || metConditions.size() > 0) {
+                conditionsMet.set(true);
+                // Perform the actual rollover
+                MetadataRolloverService.RolloverResult rolloverResult = rolloverService.rolloverClusterState(
+                    currentState,
+                    rolloverRequest.getRolloverTarget(),
+                    rolloverRequest.getNewIndexName(),
+                    rolloverRequest.getCreateIndexRequest(),
+                    metConditions,
+                    false,
+                    false
+                );
+                logger.trace("rollover result [{}]", rolloverResult);
+
+                // Update the "final" source and resulting rollover index names.
+                // Note that we use the actual rollover result for these, because
+                // even though we're single threaded, it's possible for the
+                // rollover names generated before the actual rollover to be
+                // different due to things like date resolution
+                sourceIndex.set(rolloverResult.sourceIndexName);
+                rolloverIndex.set(rolloverResult.rolloverIndexName);
+
+                // Return the new rollover cluster state, which includes the changes that create the new index
+                return rolloverResult.clusterState;
+            } else {
+                // Upon re-evaluation of the conditions, none were met, so
+                // therefore do not perform a rollover, returning the current
+                // cluster state.
+                return currentState;
+            }
+        }
+
+        @Override
+        public void onFailure(String source, Exception e) {
+            listener.onFailure(e);
+        }
+
+        @Override
+        public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+            // Now assuming we have a new state and the name of the rolled over index, we need to wait for the
+            // configured number of active shards, as well as return the names of the indices that were rolled/created
+            if (conditionsMet.get()) {
```

Comment:
Can we have a noop task here by any chance and have both states equal, do we have to account for that somehow? It seems no, but just double checking :)

Reply:
I think this be may be true if something changed between the trail rollover and actual rollover, that makes a condition go from true to false. The if statement that I will add for the previous comment, will catch this, if it happens.